src/open_inwoner/openklant/tests/test_views.py

```diff
@@ -33,6 +34,171 @@
 from .factories import KlantContactMomentAnswerFactory
 
 
+@requests_mock.Mocker()
+@override_settings(ROOT_URLCONF="open_inwoner.cms.tests.urls")
+class ContactMomentenVragenServiceTestCase(TestCase):
+    maxDiff = None
+
+    def setUp(self):
+        super().setUp()
+        MockAPIReadData.setUpServices()
+        self.service = ContactMomentenVragenService()
+        self.user = UserFactory()
+        klanten_config = OpenKlantConfig.get_solo()
+        klanten_config.exclude_contactmoment_kanalen = ["intern_initiatief"]
+        klanten_config.save()
+
+        self.contactformsubject = ContactFormSubject.objects.create(
+            subject="oip_subject",
+            subject_code="e_suite_subject_code",
+            config=klanten_config,
+        )
+
+    def test_list_questions_returns_expected_rows(self, m):
+        data = MockAPIReadData().install_mocks(m)
+        config = OpenKlantConfig.get_solo()
+
+        for user, params, expected_klantcontact, expected_contactmoment, use_rsin in (
+            (
+                data.user,
+                {"user_bsn": "100000001"},
+                data.klant_contactmoment,
+                data.contactmoment,
+                False,
+            ),
+            (
+                data.eherkenning_user,
+                {"user_kvk_or_rsin": "12345678"},
+                data.klant_contactmoment2,
+                data.contactmoment2,
+                False,
+            ),
+            (
+                data.eherkenning_user,
+                {"user_kvk_or_rsin": "12345678", "vestigingsnummer": "1234"},
+                data.klant_contactmoment4,
+                data.contactmoment_vestiging,
+                False,
+            ),
+            (
+                data.eherkenning_user,
+                # RSIN case
+                {"user_kvk_or_rsin": "000000000", "vestigingsnummer": "1234"},
+                data.klant_contactmoment4,
+                data.contactmoment_vestiging,
+                True,
+            ),
+            (
+                data.eherkenning_user,
+                # RSIN case
+                {"user_kvk_or_rsin": "000000000"},
+                data.klant_contactmoment2,
+                data.contactmoment2,
+                True,
+            ),
+        ):
+            with self.subTest(f"{user=} {params=} {use_rsin=}"):
+                config.use_rsin_for_innNnpId_query_parameter = use_rsin
+                config.save()
+                questions = list(self.service.list_questions(params, user))
+
+                detail_url = reverse(
+                    "cases:contactmoment_detail",
+                    kwargs={"kcm_uuid": uuid_from_url(expected_klantcontact["url"])},
+                )
+                self.assertEqual(
```

Comment:
Could use an additional `self.assertEqual(len(questions), 1)` to ensure that we don't have spurious stuff in there we're missing out on